### PR TITLE
Golang Homework 7.3 - Get/Post

### PR DESCRIPTION
**Pull request by stevenstr**
**Home work 7.3 - Task:**
●Web server, responds to “/” (404 on any other request)
●Support POST and GET methods
• On GET request returns HTML page, containing FORM with name and address fields + submit button.
Also there is a placeholder to display token on this page.
● On POST request reads name+address from body and creates a token as “name:address”.
Then saves this token to Cookies
●Page reloads, token value displayed on page

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	<script type="text/javascript">
+		const elem = document.querySelector("#known-token");
+		let token = document.cookie.split(';').filter((item) => item.trim().startsWith('token='))[0];
+		elem.innerHTML = token.replace('token=', '')
+	</script>
+	</html>
+`)
+
+func handleMainPage(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
```
Comment (burov) [inline main.go]: switch could improve code readability

**Review by burov — CHANGES_REQUESTED** [review]
Please use switch in handleMainPage

File: main.go (hunk 1 above)
Comment (stevenstr) [reply main.go]: Great thanks. done it)

File: main.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		w.Write(formNameAddrTmpl)
+		return
+
+	case http.MethodPost:
+		name := r.FormValue("name")
+		address := r.FormValue("address")
+		t := time.Now().Add(10 * time.Hour)
+		cookie := http.Cookie{Name: "token", Value: name + ":" + address, Expires: t}
+		http.SetCookie(w, &cookie)
+		w.Write(formNameAddrTmpl)
```
Comment (wshaman) [inline main.go:59]: It should be writing html OR redirecting.
This code will work, but could make crazy some browsers.

**Review by wshaman — APPROVED** [review]

File: main.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		http.Redirect(w, r, "/", http.StatusFound)
+		return
+	default:
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte("Not found"))
+		return
```
Comment (burov) [inline main.go:65]: return redundant because by default switches don't use ```fall trough``` behaviour

**Review by burov — APPROVED** [review]
Overall looks ok for me

**Pull request closed without merging**